File: SaneScanner/MHUICustomization+SY.m
//
//  MHUICustomization+SY.m
//  SaneScanner
//
//  Created by Stan Chevallier on 22/04/2016.
//  Copyright © 2016 Syan. All rights reserved.
//

#import "MHUICustomization+SY.h"
#import "SYGridLayout.h"
#import "SYMediaPreviewCollectionViewCell.h"

@implementation MHUICustomization (SY)

+ (MHUICustomization *)sy_defaultTheme
{
    MHUICustomization *theme = [[self alloc] init];
    [theme setMHGalleryBackgroundColor:[UIColor groupTableViewBackgroundColor]
                           forViewMode:MHGalleryViewModeImageViewerNavigationBarShown];
    [theme setShowMHShareViewInsteadOfActivityViewController:NO];
    [theme setShowOverView:YES];
    [theme setUseCustomBackButtonImageOnImageViewer:YES];
    [theme setBackButtonState:MHBackButtonStateWithBackArrow];
    [theme setOverviewTitle:@"All scans"];
    [theme setOverviewCollectionViewCellClass:SYMediaPreviewCollectionViewCell.class];
    
    SYGridLayout *layout = [[SYGridLayout alloc] init];
    [layout setMaxSize:320/3];
    [layout setMargin:2.];
    [theme setOverviewCollectionViewLayout:layout];
    
    return theme;
}

@end

File: SaneScanner/MHUICustomization+SY.swift
//
//  MHUICustomization+SY.swift
//  SaneScanner
//
//  Created by Stanislas Chevallier on 07/02/2019.
//  Copyright © 2019 Syan. All rights reserved.
//

import MHVideoPhotoGallery

extension MHUICustomization {
    static var sy_defaultTheme: MHUICustomization {
        let theme = MHUICustomization()
        theme.setMHGalleryBackgroundColor(.groupTableViewBackground, for: .imageViewerNavigationBarShown)
        theme.showMHShareViewInsteadOfActivityViewController = false
        theme.showOverView = true
        theme.useCustomBackButtonImageOnImageViewer = true
        theme.backButtonState = .withBackArrow
        theme.overviewTitle = "GALLERY OVERVIEW TITLE".localized
        theme.overviewCollectionViewCellClass = MediaPreviewCollectionViewCell.self
        
        let layout = GridLayout()
        layout.maxSize = 320/3
        layout.margin = 2
        theme.overviewCollectionViewLayout = layout
        
        MHGalleryCustomLocalizationBlock { (string) -> String? in
            let localized = string?.localized
            return string == localized ? nil : localized
        }
        
        return theme
    }
}

